packages/server/src/trpc-routers/orderbook-router.ts

```diff
@@ -0,0 +1,18 @@
+import { queryOrderbookMakerFee } from "../queries";
+import { ContractOsmoAddressSchema } from "../queries/complex/parameter-types";
+import { createTRPCRouter, publicProcedure } from "../trpc";
+
+export const orderbookRouter = createTRPCRouter({
+  getMakerFee: publicProcedure
+    .input(ContractOsmoAddressSchema.required())
+    .query(async ({ input, ctx }) => {
+      const { contractOsmoAddress } = input;
+      const { data } = await queryOrderbookMakerFee({
+        orderbookAddress: contractOsmoAddress,
+        chainList: ctx.chainList,
+      });
+      return {
+        makerFee: data,
+      };
+    }),
```

Comment:
To benefit from caching, our convention currently is to create a function in the complex folder. Then, wrap the function you want to cache in cachified.

In this case, I'd create a "orderbooks" folder under complex directory (with an index.ts), put this in "maker-fee.ts" file. And there I'd create an instance of an in memory cache and use that to wrap this in cachified. Choose a ttl that makes sense to you. This is a low overhead way to prevent some needless queries to the chain and improve user UX.

Reply:
https://github.com/osmosis-labs/osmosis-frontend/pull/3321/commits/43817c6ba60b9e5d722df68b07c77e0dda5e8755